base/essentials.jl

```diff
@@ -593,6 +593,39 @@ function unconstrain_vararg_length(va::Core.TypeofVararg)
     return Vararg{unwrapva(va)}
 end
 
+# Compute the minimum number of initialized fields for a particular datatype
+# (therefore also a lower bound on the number of fields)
+function datatype_min_ninitialized(@nospecialize t0)
```

Comment:
Nothing new from the method in `Compiler`, it was just moved to `Base` to have it available at `PartialStruct` construction